### PR TITLE
Add interpreter for ShiftRightLogicalOp

### LINKED ISSUE #1114: Add interpreter for ShiftRightLogicalOp
Following the [testing guidelines](https://github.com/openxla/stablehlo/blob/main/docs/interpreter_checklist.md)

### PR DESCRIPTION
**Pull request by ghpvnist**
Here are the constraints for the ShiftRightLogicalOp:
```
(I1) lhs is a tensor of integer type.
(I2) rhs is a tensor of integer type.
(C1) `lhs`, `rhs`, and `result` have the same type.
```
I1, I2, and C1 are covered by the ODS, so no additional tests are added.

Notes:
* Corner cases (shift overflow) has not been accounted for: #1150

closes #1114

### CODE REVIEW

**Review by sdasgup3 — APPROVED** [review]

File: docs/spec.md [hunk 1]
```diff
@@ -5010,12 +5010,14 @@ number of bits and produces a `result` tensor.
 #### Examples
 
 ```mlir
-// %lhs: [-1, -128, -36, 5, 3, 7]
-// %rhs: [1, 2, 3, 2, 1, 3]
-%result = "stablehlo.shift_right_logical"(%lhs, %rhs): (tensor<6xi8>, tensor<6xi8>) -> tensor<6xi8>
-// %result: [127, 32, 27, 1, 1, 0]
+// %lhs: [-1, 0, 1]
+// %rhs: [1, 2, 3]
+%result = "stablehlo.shift_right_logical"(%lhs, %rhs): (tensor<3xi64>, tensor<3xi64>) -> tensor<3xi64>
```
Comment (sdasgup3) [inline docs/spec.md:5015]: The input `lhs = 1` and `rhs = 3` add some value for left shift. For right shift, it would be better to have an input like shiftArithRight(16, 3) = 1.
